src/components/Dialogs/Dialog/Dialog.tsx

```diff
@@ -1,18 +1,24 @@
 import * as React from 'react';
+import { MdClose } from 'react-icons/md';
 import { bem } from '../../../utils';
-import { Heading } from '../../Heading';
 import { Button } from '../../Buttons';
 import { Modal, ModalProps } from '../../Modal';
 import styles from './Dialog.scss';
+import { Heading } from '../../Heading';
+import { ButtonContext } from '../../../constants/component-specific';
 
 export type ButtonProps = {
     onClick: () => void;
     label: string;
 };
 
 export interface DialogProps extends ModalProps {
+    /** the component which will trigger the dialog */
+    dialogTrigger?: React.ReactNode;
```

Comment:
I believe this is not needed, it can work as before. We have the trigger outside and it can be anything. The dialog opens based on the isOpen.